static/js/editor.ts

```diff
@@ -0,0 +1,106 @@
+export type Breakpoints = Record<number, string>;
+
+
+export interface HedyEditorCreator {
+  // TODO: Not sure yet if it should return a HedyEditor or change local variable names
+  /**
+   * This function should initialize the editor and set up all the required
+   * event handlers
+   * @param $editor reference to the div that contains the main editor
+   */
+  initializeMainEditor: ($editor: JQuery) => HedyEditor | undefined;
+  /**
+   * 
+   * @param element the element that will contain this editor
+   * @param isReadOnly to decide weather to remove the cursor
+   * @param isMainEditor should we show the line numbers
+   */
+  turnIntoEditor: (element: HTMLElement, isReadOnly: boolean, isMainEditor: boolean) => HedyEditor;
+
+  /**
+   * Ininitialize an editor that appears in a modal
+   * @param $editor reference to the div that contains this editor
+   */
+  initializeModalEditor: ($editor: JQuery) => HedyEditor;
+}
+
+export interface HedyEditor {
+  /**
+ * Represents whether there's an open 'ask' prompt
+ */
+  askPromptOpen: boolean;
```

Comment:
I thought a bit about whether to include this variable in the interface or not. This variable is used in the main editor to focus on the ask modal, and other things. I included here, because I think that aspects that control HTML of the page that contains the editor and also control behaviors of what one would think is the editor, should go in this interface.